app/classes/Transvision/Utils.php

```diff
@@ -438,10 +438,31 @@ public static function ago($datetime, $ref_time = '')
      * This is used on views which also exist in our public API
      * https://github.com/mozfr/transvision/wiki/JSON-API
      *
-     * @return string URL with 'json' appended as part of the query string
+     * @param  boolean $revert if true, invert the locale and the source_locale in the API generated link
+     * @return string  URL with 'json' appended as part of the query string
      */
-    public static function redirectToAPI()
+    public static function redirectToAPI($revert = false)
     {
-        return $_SERVER["REQUEST_URI"] . (is_null($_SERVER['QUERY_STRING']) ? '?json' : '&json');
+        if (! is_null($_SERVER['QUERY_STRING'])) {
+            // Allow to revert the locale and the sourcelocale when redirecting to the API
+            if ($revert) {
+                $arg = [];
+                parse_str($_SERVER['QUERY_STRING'], $arg);
+                $sourcelocale = $arg['sourcelocale'];
+                $arg['sourcelocale'] = $arg['locale'];
+                $arg['locale'] = $sourcelocale;
+                $query = http_build_query($arg);
```

Comment:
That whole part for swapping locales is ok but I don't think you should build the query at that time